.npm/postinstall.js

```diff
@@ -1,10 +1,76 @@
-if (!process.env.CI) {
-  const { spawnSync } = require('child_process');
-  const { getExePath } = require('./get-exe');
+const { spawnSync } = require("child_process")
 
+async function install() {
+  if (process.env.CI) {
+    return
+  }
+  const exePath = await downloadBinary()
+  if (!(["win32", "cygwin"].includes(process.platform))) {
+    const { chmodSync } = require("fs")
+    chmodSync(exePath, "755")
+  }
   // run install
-  spawnSync(getExePath(), ['install', '-f'], {
+  spawnSync(exePath, ["install", "-f"], {
     cwd: process.env.INIT_CWD || process.cwd,
-    stdio: 'inherit',
-  });
+    stdio: "inherit",
+  })
 }
+
+function getDownloadURL() {
+  // Detect OS
+  // https://nodejs.org/api/process.html#process_process_platform
+  let goOS = process.platform
+  let extension = ""
+  if (["win32", "cygwin"].includes(process.platform)) {
+    goOS = "windows"
+    extension = ".exe"
+  }
+
+  // Convert the goOS to the os name in the download URL
+  let downloadOS = goOS === "darwin" ? "macOS" : goOS
+  downloadOS = `${downloadOS.charAt(0).toUpperCase()}${downloadOS.slice(1)}`
+
+  // Detect architecture
+  // https://nodejs.org/api/process.html#process_process_arch
+  let arch = process.arch
+  switch (process.arch) {
+    case "x64": {
+      arch = "x86_64"
+      break
+    }
+    case "x32":
+    case "ia32": {
+      arch = "i386"
+      break
+    }
+  }
+  const version = require("./package.json").version
+
+  return `https://github.com/evilmartians/lefthook/releases/download/v${version}/lefthook_${version}_${downloadOS}_${arch}${extension}`
+}
+
+const { DownloaderHelper } = require("node-downloader-helper")
+const path = require("path")
+
+async function downloadBinary() {
+  // TODO zip the binaries to reduce the download size
+  const downloadURL = getDownloadURL()
+  const extension = ["win32", "cygwin"].includes(process.platform) ? ".exe" : ""
+  const fileName = `lefthook${extension}`
+  const binDir = path.join(__dirname, "bin")
+  const dl = new DownloaderHelper(downloadURL, binDir, {
+    fileName,
+    retry: { maxRetries: 5, delay: 50 },
+  })
+  dl.on("error", (e) => {
+    e.message = `Failed to download ${fileName}: ${e.message} while fetching ${downloadURL}`
+    throw e
```

Comment:
Does throwing the error here prevents retrying? What I had here allowed the code to retry downloading.

Reply:
Oh, this is good question.

Reply:
Looks like it actually prevent retrying. But I believe [it is a bug](https://github.com/hgouveia/node-downloader-helper/issues/57).

Reply:
I think we should revert this change. What I had here allowed retrying, and also printed the actual error message.

Reply:
Well, the problem is that printing doesn't work (at least with yarn). I will experiment with error throwing again later.

Reply:
You could replace this `throw e` with `console.error(e)`, and then revert what I had to throw the last error.
```suggestion
    console.error(e)
```

Reply:
`console.error` output isn't visible when yarn is used to install package.

Reply:
Then there is no way to retry with Yarn. I recommend throwing the last error for those on Yarn and let the algorithm work normally for others.

Reply:
Reverted my commit

I couldn't make yarn to print error message in a descriptive way I want because yarn is exiting on the first `error` event thrown (and I insist that this is caused by the bug in node-download helper, see https://github.com/hgouveia/node-downloader-helper/issues/57#issuecomment-848817013).

I give up. Let it be so.